src/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs

```diff
@@ -2889,10 +2894,28 @@ public int Compare(object left, object right)
             }
         }
 
+        [TypeDescriptionProvider(typeof(ComNativeDescriptorProxy))]
```

Comment:
Would specifying the WinForms type directly here work? Like:

`[TypeDescriptionProvider("System.Windows.Forms.ComponentModel.Com2Interop.ComNativeDescriptor, System.Windows.Forms")]`

Reply:
Yes, it probably can. My intent here was to make this easier to debug and determine what is failing. I find debugging these kind of attributes very cumbersome so I prefer wrapping them in simple proxies when possible. Is there a strong preference?

Reply:
> debugging these kind of attributes

I am not sure why the proxies make debugging easier. You should be able to just stop on first chance exceptions to get the initial idea of where it is failing, no?

Reply:
> Is there a strong preference?

Other places in CoreFX just use this attribute with the name, e.g. https://github.com/dotnet/corefx/blob/master/src/System.Private.Xml.Linq/src/System/Xml/Linq/XAttribute.cs#L20

Reply:
>You should be able to just stop on first chance exceptions to get the initial idea of where it is failing, no?

Yep, but then I can't try doing other operations in the debugger using the immediate window. Basically the runtime is doing too much and if I want to investigate the visible types or try a different assembly name, I need to recompile. This approach allows investigation rather than just "oh it isn't there. Let me recompile and try something else".

>Other places in CoreFX just use this attribute with the name

Sounds good. I will change it to that. Hopefully this will be a short lived scenario and the actual types can be brought into this assembly.